### ISSUE TITLE
how to get file path or bytes?

### ISSUE THREAD

**Issue by linguitang** [Sh1d0w/multi_image_picker#42]
after get images, I want to upload to server.but, how to get file path or bytes?

**Comment by Sh1d0w** [comment]
Hi, you can get the original image data and pass it as ByteData to the http request like this:

```dart

    // string to uri
    Uri uri = Uri.parse('$_apiEndpoint/some/path');

    // create multipart request
    MultipartRequest request = http.MultipartRequest("POST", uri);

    List<int> byteData = await asset.requestOriginal();
    MultipartFile multipartFile = MultipartFile.fromBytes(
      'photo',
      byteData,
      filename: 'some-file-name.jpg',
      contentType: MediaType("image", "jpg"),
    );

    // add file to multipart
    request.files.add(multipartFile);
    // send
    var response = await request.send();
```

I will add example for this in the readme as well.

**Comment by linguitang** [comment]
the result of "await asset.requestOriginal() " is type of _ByteDataView

**Comment by Sh1d0w** [comment]
Whoops, it should be 

```dart
ByteData imageData = await asset.requestOriginal();
List<int> byteData = imageData.buffer.asUint8List();
```

Don't forget to release the data after you upload it from the asset:

```dart
asset.requestOriginal();
```

**Comment by linguitang** [comment]
I also want to know if it can support the simultaneous selection of pictures and videos from albums in the future.

**Comment by Sh1d0w** [comment]
Currently you can't access the image name via the Asset class, but you can compare identifiers, which are unique, or generate some name based on them.

Regarding videos - I do not plan to add this my only focus at the moment are making the plugin best image picker available for Flutter, but it is possible to allow picking videos as well. A PR implementing this is more than welcome.

**Comment by ouardia-y** [comment]
After following you example, I have a List<Asset> images of which I want to retrieve the data and be able to upload it to firebase storage. With image picker (plugin) it returns a file and I can upload that automatically. I tried looping through the list of images but I can't convert it to a file. Any ideas on how to proceed?

**Comment by Sh1d0w** [comment]
@ouardia-y According to Firestore docs, you can pass either a File or Binary data directly:

https://pub.dartlang.org/documentation/firebase_storage/latest/firebase_storage/StorageReference/putData.html

I've updated the docs to showcase how to do this, see here:

https://sh1d0w.github.io/multi_image_picker/#/faq?id=how-to-upload-images-to-firebase

**Comment by ouardia-y** [comment]
Thank you so much! I didn't see it in the documentation earlier on. I really appreciate your work!

**Comment by Sh1d0w** [comment]
@ouardia-y Thank you very much. I see, it may be harder to see the documentation link initially, so I've just added a whole section into the README that points out to the documentation.

The docs are also still a work in progress, I am trying to improve them with a lot of examples, so I am occasionally moving stuff around. 

The example for firebase upload is now moved out of the FAQ section in a more visible place:

https://sh1d0w.github.io/multi_image_picker/#/upload?id=firebase

Thanks for using the plugin and the feedback!
